### ISSUE TITLE
After Kubernetes Update longhorn-csi-plugin and longhorn-driver-deployer pods dosnt get up

### ISSUE THREAD

**Issue by corx96** [longhorn/longhorn#1036]
Hello @yasker ,

Specs:
Longhorn: v.0.5.0
Rancher: v2.3.5
Cubernetes: v1.17.2

After updating our cluster to Kubernetes v.1.17.2 longhorn does not work anymore. The cluster consists of one node.

Here are some logs etc.

`kubectl get pods -n longhorn-system`

> NAME                                                  READY   STATUS             RESTARTS   AGE
csi-attacher-769b87d76d-559tp                         0/1     CrashLoopBackOff   61         5h42m
csi-attacher-769b87d76d-5gz7g                         0/1     CrashLoopBackOff   61         5h42m
csi-attacher-769b87d76d-fx7fx                         0/1     CrashLoopBackOff   61         5h42m
csi-provisioner-5bd8d7cb74-q2xtt                      1/1     Running            7          176d
csi-provisioner-5bd8d7cb74-r7pmg                      1/1     Running            8          176d
csi-provisioner-5bd8d7cb74-wppkq                      1/1     Running            8          176d
engine-image-ei-9bea8a9c-2vwzt                        1/1     Running            6          176d
longhorn-csi-plugin-f7l7w                             0/2     CrashLoopBackOff   157        5h42m
longhorn-driver-deployer-5d9c76865f-6pc2t             0/1     Init:0/1           0          48m
longhorn-driver-deployer-7f6548697d-vp4hm             0/1     Init:0/1           0          48m
longhorn-manager-7jx9c                                1/1     Running            0          47m
longhorn-ui-867dddf67c-wx92h                          1/1     Running            14         176d

> 
The csi-attachers run again and again and then go out again

Log from longhorn-manager:

> E0129 13:50:52.137990 1 reflector.go:134] k8s.io/client-go/informers/factory.go:131: Failed to list *v1beta2.DaemonSet: the server could not find the requested resource
E0129 13:50:53.139100 1 reflector.go:134] k8s.io/client-go/informers/factory.go:131: Failed to list *v1beta2.DaemonSet: the server could not find the requested resource
E0129 13:50:54.140200 1 reflector.go:134] k8s.io/client-go/informers/factory.go:131: Failed to list *v1beta2.DaemonSet: the server could not find the requested resource
E0129 13:50:55.141182 1 reflector.go:134] k8s.io/client-go/informers/factory.go:131: Failed to list *v1beta2.DaemonSet: the server could not find the requested resource
....
> 

Log from longhorn-csi-plugin:

> time="2020-01-29T13:51:03Z" level=info msg="CSI Driver: io.rancher.longhorn csiVersion: 0.2.0"
time="2020-01-29T13:51:03Z" level=fatal msg="Error starting CSI manager: Failed to initialize Longhorn API client: Get http://longhorn-backend:9500/v1: dial tcp 10.43.151.106:9500: connect: connection refused"
> 
log from driver-registrar:

> I0129 13:52:01.079032 1 main.go:138] Version: v0.4.2-0-gc1189c43
I0129 13:52:01.079174 1 main.go:145] Attempting to open a gRPC connection with: "/var/lib/kubelet/plugins/io.rancher.longhorn/csi.sock"
I0129 13:52:01.079191 1 connection.go:69] Connecting to /var/lib/kubelet/plugins/io.rancher.longhorn/csi.sock
I0129 13:52:01.079577 1 connection.go:96] Still trying, connection is CONNECTING
I0129 13:52:01.079762 1 connection.go:96] Still trying, connection is TRANSIENT_FAILURE
I0129 13:52:02.079833 1 connection.go:96] Still trying, connection is TRANSIENT_FAILURE
I0129 13:52:03.121780 1 connection.go:96] Still trying, connection is TRANSIENT_FAILURE
...
I0129 13:53:01.079337 1 connection.go:89] Connection timed out
I0129 13:53:01.079370 1 main.go:153] Calling CSI driver to discover driver name.
I0129 13:53:01.079390 1 connection.go:137] GRPC call: /csi.v0.Identity/GetPluginInfo
I0129 13:53:01.079396 1 connection.go:138] GRPC request: {}
I0129 13:53:01.082357 1 connection.go:140] GRPC response: {}
I0129 13:53:01.082735 1 connection.go:141] GRPC error: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
E0129 13:53:01.082749 1 main.go:158] rpc error: code = Unavailable desc = all SubConns are in TransientFailure
> 

Any idea how to fix it?

**Comment by yasker** [comment]
@corx96 Longhorn v0.5.0 doesn't support Kubernetes v1.17.2, there are Kubernetes API group changes that break the backward compatibility.

Can you upgrade to v0.6.2 instead? Notice that the offline upgrade is needed. See https://github.com/longhorn/longhorn/blob/master/docs/upgrade.md#upgrade-from-older-versions-to-v062  for details.

Don't upgrade to v0.7.0 directly, we don't support upgrading from v0.5.0 to v0.7.0 directly. After upgrade to v0.6.2, you can follow the direction at https://github.com/longhorn/longhorn/blob/master/docs/upgrade-from-v0.6.2-to-v0.7.0.md to upgrade from v0.6.2 to v0.7.0.

**Comment by corx96** [comment]
@yasker thanks for your fast answer!

i will try to upgrade longhorn.

Do you think its possible to detach the volumes without rancher working fine ?

**Comment by yasker** [comment]
If you meant by executing the offline upgrade, you can upgrade Longhorn first, then shutdown workload/detach the volume and upgrade the Longhorn engine.

**Comment by corx96** [comment]
@yasker thank you very much, an update from v0.5.0 to v0.6.2 and then to v0.7.0 worked for me.

 If anyone else has the problem, I would like to describe my solution briefly to perhaps save some time for someone.

1. try to update to v0.6.2
`This resulted in the following error message: 
Failed to install app longhorn-system. Error: UPGRADE FAILED: failed decoding reader into objects: [unable to recognize "": no matches for kind "DaemonSet" in version "apps/v1beta2", unable to recognize "": no matches for kind "Deployment" in version "apps/v1beta2"] `
Which I solved this problem with this https://github.com/longhorn/longhorn/issues/887#issuecomment-562355004  instructions
2. Then i stopped my workloads and updated the engine images to v0.6.2
3. Then I tried to start my pods which did not work with error msg
`driver name io.rancher.longhorn not found in the list of registered CSI drivers`
I tried the fixes mentioned here https://github.com/longhorn/longhorn/issues/455#issuecomment-480008806 but none of them worked out for me
4. Then i updated to v0.7.0 
5. Stopped my workloads and updated the Engine Images to v0.7.0.
6. Started my Workloads.

After that everything worked as intended.
